src/main/java/com/amazonaws/encryptionsdk/model/CiphertextHeaders.java

```diff
@@ -179,6 +180,9 @@ public Boolean isComplete() {
      */
     private int parseVersion(final byte[] b, final int off) throws ParseException {
         version_ = PrimitivesParser.parseByte(b, off);
+        if(CiphertextType.deserialize(version_).getValue() != Byte.parseByte(VersionInfo.VERSION_NUM, 16)) {
```

Comment:
This is the major version of this implementation, not the ciphertext message format version. You want `CURRENT_CIPHERTEXT_VERSION`.
```suggestion
        if(CiphertextType.deserialize(version_).getValue() != Byte.parseByte(VersionInfo.CURRENT_CIPHERTEXT_VERSION, 16)) {
```

Reply:
Please actually change as follows: (note lack of parsing and space after `if`
```java
         if (CiphertextType.deserialize(version_).getValue() != VersionInfo.CURRENT_CIPHERTEXT_VERSION) { 
```